wasm.go

```diff
@@ -207,6 +210,12 @@ func (r *runtime) InstantiateModuleWithConfig(code *CompiledCode, config *Module
 	}
 
 	module := config.replaceImports(code.module)
+	if module != code.module {
+		// IF the module changed, we have to Compile again before instantiation.
```

Comment:
only question is why we have to if we are only changing extern string names.. are the strings embedded in the code somehow?

```suggestion
		// If replacing imports had an effect, the module changed, so we have to recompile it.
```

Reply:
maybe we might needed a "TODO: implement re-linking" as probably this is more about where the imported functions point to vs re-compiing the code sections, right?

Reply:
good question.. the compiled native code itself is not coupled with names, but the function instances will be different between with and without re-exports

Reply:
I think we should move this replacing config into `CompileModule`

Reply:
https://github.com/tetratelabs/wazero/pull/469/commits/478b41b3d485e669cd6cad918ef3bc4ee2701b1c added todo

Reply:
oh well yeah it should be possible to reuse the code actually.. lemme fix

Reply:
https://github.com/tetratelabs/wazero/pull/469/commits/42965e243879699d1e3b6a26578959231918f973 fixed!

Reply:
hmm realized that we use `*wasm.Module` as a cache key so this makes it impossible to share code.. let me revert it and resolve it in another pr by introducing `ID` for wasm.Module